x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/entsearch/engine/action/PostEngineAction.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.entsearch.engine.action;
+
+import org.elasticsearch.action.ActionType;
+
+public class PostEngineAction extends ActionType<PutEngineAction.Response> {
```

Comment:
I started using the same Request/Response as `PutEngineAction` since I really just want to return a 409 if it exists otherwise everything should be the same. If we decide to go this route we can refactor the request/response to something that's shared.